src/editor/extended_python_generator.ts

```diff
@@ -31,6 +31,8 @@ import * as commonStorage from '../storage/common_storage';
 export class ExtendedPythonGenerator extends PythonGenerator {
   private currentModule: commonStorage.Module | null = null;
   private mapWorkspaceIdToExportedBlocks: { [key: string]: Block[] } = Object.create(null);
+  protected methods_: {[key: string]: string} = Object.create(null);
```

Comment:
Why protected instead of private?
Why the trailing _?

Reply:
I almost always go for protected instead of private so it can be subclassed later.
The trailing _ was because I was copying the style from `definitions_`

Reply:
I'm the opposite. I go for the most restrictive visibility first and then make it more visible only if necessary.
If it needs to be subclassed later, it can be changed from private to protected then.
https://google.github.io/styleguide/tsguide.html#visibility

Reply:
While I agree fully about not making things public by default (as the style guide says), protected seems to be better than private.   Imagine the mess we would be in if Blockly had made everything private instead of protected.    (I'll admit that this tendency of mine is because I am almost always creating libraries or systems instead of applications.)

Reply:
That makes sense for libraries where you (as author of the library) want to allow a class to be subclassed. In this case, it is not a library and within this application, we don't currently want to subclass ExtendedPythonGenerator.

I'm going to approve (and merge) this PR so we can keep making progress.